src/asInput/index.jsx

```diff
@@ -117,6 +118,7 @@ const asInput = (WrappedComponent, labelFirst = true) => {
   NewComponent.defaultProps = {
     onChange: () => {},
     onBlur: () => {},
+    id: undefined,
```

Comment:
if we want to default to `newId('asInput')` should this also be the default prop value?

Reply:
Reading this made me realize -- If I set the default value here, would I even need to do the check above?

Reply:
Yes, because this only fills in the `id` value if somebody doesn't pass in the `id` prop, but if somebody passes in an `undefined` value for the `id` prop we probably want to default to the `newId` id, right?

Reply:
Ah yes, that is indeed an edge case. I'll update this default and change to the || above.

Reply:
Actually, I'm not sure about this after all - I'm currently investigating.

Reply:
Sorry @fysheets I might have led you astray - check for yourself, but I think the `defaultProps` might also coerce `undefined` values.

The only edge case is that [React treats `undefined` and `null` prop values differently](https://github.com/facebook/react/issues/2166) - it *won't* coerce `null` - do you think it makes sense to create a default id value if the prop value is `null`? If so, then we should keep the previous check.